### PR TITLE
Enable "Showing XX rows" for all types of visualizations

### LINKED ISSUE #29261: Enable "Showing XX rows" not only for table visualisations
**Is your feature request related to a problem? Please describe.**
When having run a query and the visualisation is a table the lower bar looks like this
Screenshot 2023-03-16 at 09 03 05
and is showing the number of returned rows.

When switching to any other kind of visualisation the the element showing the returned rows is not showing. 
Screenshot 2023-03-16 at 09 03 40
This is valid for the table and the visualisation view.
Screenshot 2023-03-16 at 09 03 52

**Describe the solution you'd like**
For any time that you are in the table view where the query outputs are shown, please add the indication of how many rows are returned.

**Describe alternatives you've considered**
Always show the returned rows, no matter if in table or visualisation view.

**How important is this feature to you?**
Its a quality of life improvement to quickly understand the query results.

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes https://github.com/metabase/metabase/issues/29261

### Description

The goal is to show the "Showing XX rows" label for all types of visualizations, not only table visualizations. 

There were no tests to update and I didn't think this change deserved a new one - please let me know if you disagree.

#### Before

https://github.com/metabase/metabase/assets/6830683/181aaca1-0bc5-409d-aa92-1247c569d207

#### After

https://github.com/metabase/metabase/assets/6830683/68c51487-9d4f-4519-bf75-64f4222c86bd

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30843)

### CODE REVIEW

**Review by ranquild — COMMENTED** [review]
@kamilmielnik looks good! could you also add tests for your change? You could add unit tests in `QuestionRowCount.unit.spec.tsx` for `shouldRender`

**Comment by kamilmielnik** [comment]
> You could add unit tests in QuestionRowCount.unit.spec.tsx for shouldRender

I can imagine that this test would be checking whether the result of `shouldRender` is `true` regardless of `question.display()`. But since this PR removes the `question` argument from `shouldRender`, it renders such a test redundant.

@ranquild  Am I missing something, or maybe did you have a different test case in mind?

**Comment by kamilmielnik** [comment]
The only idea I had here for a valid test case was to create a new test suite for `query_builder/components/view/ViewFooter.jsx` (or higher up in the hierarchy) and check whether the "Showing XX rows" message is rendered regardless of `question.display()` - because there is `question` there.
But it'd be "expensive" to add a new test suite just for that. Please let me know if you think the benefit would justify the "cost" here.

**Comment by ranquild** [comment]
@kamilmielnik there is existing test suite in `QueryBuilder.unit.spec.tsx`, and it should be possible to add tests for this change there. But before doing that, could you talk to @oisincoveney? He has modified this test suite heavily in on of his integration branches.

**Comment by oisincoveney** [comment]
> @kamilmielnik there is an existing test suite in `QueryBuilder.unit.spec.tsx`, and it should be possible to add tests for this change there. But before doing that, could you talk to @oisincoveney? He has modified this test suite heavily in one of his integration branches.

Yep, I'm working on a `QueryBuilder.unit.spec.tsx` within the `29421-integration-branch` (#30768)  - it's pending a final approval from @mazameli, so if we can get it merged tonight you would be able to use it, or if you want you can also take the skeleton (i.e. the `setup` function) and work with that. It's your call - I think it would be better to wait for the merge to save your own sanity from any conflicts

**Comment by kamilmielnik** [comment]
> It's your call - I think it would be better to wait for the merge to save your own sanity from any conflicts

Got it, thanks!

> @kamilmielnik there is an existing test suite in `QueryBuilder.unit.spec.tsx`, and it should be possible to add tests for this change there. 

@ranquild Can you please describe the test case(s) you have in mind?

**Comment by ranquild** [comment]
@kamilmielnik render the QueryBuilder with 2 different cards (= questions) and questions results and assert that row count is rendered. One card would be with `display: "table"`, another one with something else, like `display: "line"`, which is the most popular visualization type we have in the app.

File: frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx
Comment (ranquild) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx:304]: We have `setupXXX` calls inside `setup` function

File: frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx
Comment (ranquild) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx:303]: `it.each` to it would work with IDEs

File: frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx
Comment (ranquild) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx:298]: nit: I'd probably add constants like `TEST_CARD_LINE` or whatever and not create them here, but no strong opinion about this

File: frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx]: e0f6d0eb88d222829720070d77261c2491fad2cb

**Review by ranquild — APPROVED** [review]

**Pull request merged**
